HTTP methods docs and REST example (#8108)
zeit/next.js: packages/next/README.md

- `res` refers to [NextApiResponse](https://github.com/zeit/next.js/blob/v9.0.0/packages/next-server/lib/utils.ts#L168-L178) which extends [http.ServerResponse](https://nodejs.org/api/http.html#http_class_http_serverresponse)

> To handle different HTTP methods with your api calls you can handle it by `req.method` in your resolver function:

Reviewer: ```suggestion
To handle different HTTP methods for API calls you can access `req.method` in your resolver function:
```
